core/src/Kokkos_Array.hpp

```diff
@@ -387,4 +454,11 @@ struct Array<T, KOKKOS_INVALID_INDEX, Array<>::strided> {
 
 }  // namespace Kokkos
 
+namespace std {
```

Comment:
wait why are we doing namespace std?

Reply:
@crtrott , this isn't uncommon, providing specializations of `namespace std` templates that interact with your own types. It's worth a conversation over whether we want to consistently do this (are there other things which need `std` specializations?), whether we always want to have a `Kokkos` mirror (`Kokkos::tuple_size`) or both, but this isn't atypical.

Reply:
This is the only way to specialize `std::tuple_size`, which is needed for structured bindings support.  See https://en.cppreference.com/w/cpp/language/structured_binding. It's standard practice to do this for tuple-like or pair-like types that the user expects to be able to destructure.